docs(wasm): support http(s) fetch of Wasm files (#3625)
dapr/docs: daprdocs/content/en/reference/components-reference/supported-bindings/wasm.md

loads `rewrite.wasm` from the current directory of the process. On Kubernetes,
see [How to: Mount Pod volumes to the Dapr sidecar]({{< ref kubernetes-volume-mounts.md >}})
to configure a filesystem mount that can contain Wasm binaries.

> It is also possible to fetch the wasm binary from a remote URL. In this case,

Reviewer: ```suggestion
It is also possible to fetch the Wasm binary from a remote URL. In this case,
```

> the URL must point exactly to one wasm binary, e.g.:

Reviewer: ```suggestion
the URL must point exactly to one Wasm binary. For example: 
```

> `http://example.com/rewrite.wasm` or `https://example.com/rewrite.wasm`. 

Reviewer: ```suggestion
- `http://example.com/rewrite.wasm`, or 
- `https://example.com/rewrite.wasm`. 
```

- `args` any CLI arguments, comma-separated. This excludes the program name.

For example, consider binding the `url` to a Ruby interpreter, such as from
[webassembly-language-runtimes](https://github.com/vmware-labs/webassembly-language-runtimes/releases/tag/ruby%2F3.2.0%2B20230215-1349da9):

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: wasm
spec:
  type: bindings.wasm
  version: v1
  metadata:
  - name: url
    value: "https://github.com/vmware-labs/webassembly-language-runtimes/releases/download/ruby%2F3.2.0%2B20230215-1349da9/ruby-3.2.0-slim.wasm"
```

> and assuming that you started your Wasm Binding at port 3500:

Reviewer: ```suggestion
Assuming that you started your Wasm Binding at port 3500, you'd run:
```

```
$ dapr run --app-id wasm --dapr-http-port 3500 --resources-path components
```

> then, the following request will respond back with "Hello, salaboy":

Reviewer: ```suggestion
The following request responds with "Hello, salaboy":
```

Reviewer 2: One more thing, since you have changed the args, the respond should be "Hello salaboy".

Author: it's more precisely `Hello "salaboy"`

- `args` any CLI arguments, comma-separated. This excludes the program name.

For example, consider binding the `url` to a Ruby interpreter, such as from
[webassembly-language-runtimes](https://github.com/vmware-labs/webassembly-language-runtimes/releases/tag/ruby%2F3.2.0%2B20230215-1349da9):

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: wasm
spec:
  type: bindings.wasm
  version: v1
  metadata:
  - name: url
    value: "https://github.com/vmware-labs/webassembly-language-runtimes/releases/download/ruby%2F3.2.0%2B20230215-1349da9/ruby-3.2.0-slim.wasm"

> ```

Author: I was not able to test this snippet though (see discord thread)

Author: I have updated the documentation to actually reflect the implementation. The issue is the original example was tokenizing naively at `,`, causing the following arglist to be split incorrectly:

```
    "args": "-ne,'print \"Hello, \"; print'" => [-ne] ['print "Hello] [ "; print]
```

a working example is instead:

```
    "args": "-ne,print \"Hello \"; print" => [-ne] [print "Hello "; print]
```

notice how quotes are currently being ignored. This is obviously suboptimal and should be fixed :)

- `args` any CLI arguments, comma-separated. This excludes the program name.

For example, consider binding the `url` to a Ruby interpreter, such as from
[webassembly-language-runtimes](https://github.com/vmware-labs/webassembly-language-runtimes/releases/tag/ruby%2F3.2.0%2B20230215-1349da9):

```yaml
apiVersion: dapr.io/v1alpha1
kind: Component
metadata:
  name: wasm
spec:
  type: bindings.wasm
  version: v1
  metadata:
  - name: url
    value: "https://github.com/vmware-labs/webassembly-language-runtimes/releases/download/ruby%2F3.2.0%2B20230215-1349da9/ruby-3.2.0-slim.wasm"
```

> Assuming that you started your Wasm Binding at port 3500, you'd run:

Reviewer 2: Assuming that you started your Dapr at port 3500 with the Wasm Binding, you'd run:

Author: I am rephrasing because now that I am re-reading that, the phrasing was awkward to begin with:

> Assuming that you wanted to start your Dapr at port 3500 with the Wasm Binding, you'd run:
